### ISSUE TITLE
@Size causing java.lang.OutOfMemoryError: Java heap space

### ISSUE THREAD

**Issue by Redid** [benas/random-beans#289]
Hi random-beans Team,

`@Size` on `String` is causing `java.lang.OutOfMemoryError: Java heap space`.
This bug is not always happening, but during debugging there are always problems.

Example:
```
public class TestString {
    @Size(min = 2)
    public String str1;
}
```

Also it is not working at all with validation-api pre 2.0, even if `Java heap space` doesn't appear `@Size` is just ignored. 

Best regards,
Redid

**Comment by benas** [comment]
> This bug is not always happening

Yeah, with random beans, we have random bugs 😄 Sorry, I couldn't resist on this..

Anyway, thank you for pointing this out!
Will take a look at this and get back to you asap

**Comment by benas** [comment]
Hi,

I'm not able to reproduce the issue even in a debug session.

Here is the type:

```java
public class Person {

    @Size(min = 50, max = 100)
    private String name;

    public String getName() {
        return name;
    }

}
```

And the (passing) test:

```java
    @Test
    void testSizeOnString() {
        Person person = EnhancedRandom.random(Person.class);
        Assertions.assertThat(person).isNotNull();
        Assertions.assertThat(person.getName().length()).isBetween(50, 100);
    }
```

> Also it is not working at all with validation-api pre 2.0

The previous test uses:

```xml
<validation-api.version>1.1.0.Final</validation-api.version>
<hibernate-validator.version>5.4.1.Final</hibernate-validator.version>
```

and it passes. Can you provide the version of the API and implementation you use ?
